debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/AbstractMysqlFieldReader.java

```diff
@@ -49,10 +56,17 @@ else if (column.jdbcType() == Types.TINYINT || column.jdbcType() == Types.SMALLI
         // DBZ-2673
         // It is necessary to check the type names as types like ENUM and SET are
         // also reported as JDBC type char
-        else if ("CHAR".equals(column.typeName()) ||
-                "VARCHAR".equals(column.typeName()) ||
-                "TEXT".equals(column.typeName())) {
-            return rs.getBytes(columnIndex);
+        else if (connectorConfig != null && !connectorConfig.customConverterRegistry().isEmpty() &&
```

Comment:
I think this all needs a comment in the code: why are doing the specific handling in this situation? The one above isn't enough, as it just states why type names are checked instead of ids, but it's lacking the motivation for the entire routine to begin with. Also let's create a constant `Set<String>` with the affected column names, using a descriptive name for its purpose. This will also help to make sense out of all this.

Reply:
@gunnarmorling Custom converter is operating on raw data coming from database so not only data types mapping is not consistent accross connectors but even for the same connector you can receive different classes for the same logical type depending if you are in snapshot or streaming phase. Hence the `JdbcValueConverter` et al hav thos `instanceof` ifs.

There was an issue DBZ-2673 reported that hinted that for customonverter it would be nice to get `bytes[]` for both snapshot and streaming phase as snapshot phase was providing `String`.
Unfortunately that has broken non-UTF databases as the snapshot was converting the string fields using wrong character set.

So the way out is - ether this PR, that would use the String as the usual approach and with custom converters it tries to convert to bytes with risk that some exotic character sets will not be supported in which case the fallback is available.

The other option would be not to use {{UTF-8}} as default for connecting to database. But that sounded to me like a very risky and too drastic change.